statusHandler/termuic/termuiRenders/widgetsRender.go

```diff
@@ -0,0 +1,217 @@
+package termuiRenders
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	ui "github.com/gizak/termui/v3"
+	"github.com/gizak/termui/v3/widgets"
+)
+
+const completRow = 1.0
+const completCol = 1.0
+
+const numLogLinesSmallLog = 10
+
+//WidgetsRender will define termui widgets that need to display a termui console
+type WidgetsRender struct {
+	grid       *ui.Grid
+	lLog       *widgets.List
+	pPublicKey *widgets.Paragraph
+	pShardId   *widgets.Paragraph
+
+	pCountLeader         *widgets.Paragraph
+	pCountConsensus      *widgets.Paragraph
+	pCountAcceptedBlocks *widgets.Paragraph
+
+	pTxPoolLoad        *widgets.Paragraph
+	pNumConnectedPeers *widgets.Paragraph
+
+	tSyncInfo *widgets.Table
+
+	lInstanceInfo        *widgets.List
+	lChainInfo           *widgets.List
+	termuiConsoleMetrics *sync.Map
+}
+
+//NewWidgetsRender method will create new WidgetsRender that display termui console
+func NewWidgetsRender(metricData *sync.Map, grid *ui.Grid) *WidgetsRender {
+	self := &WidgetsRender{
+		termuiConsoleMetrics: metricData,
+		grid:                 grid,
+	}
+	self.initWidgets()
+	self.setGrid()
+
+	return self
+}
+
+func (wr *WidgetsRender) initWidgets() {
+	wr.pPublicKey = widgets.NewParagraph()
+	wr.pShardId = widgets.NewParagraph()
+
+	wr.pCountLeader = widgets.NewParagraph()
+	wr.pCountConsensus = widgets.NewParagraph()
+	wr.pCountAcceptedBlocks = widgets.NewParagraph()
+
+	wr.pTxPoolLoad = widgets.NewParagraph()
+	wr.pNumConnectedPeers = widgets.NewParagraph()
+
+	wr.lLog = widgets.NewList()
+
+	wr.tSyncInfo = widgets.NewTable()
+	wr.tSyncInfo.Rows = [][]string{
+		{"", "", "", ""},
+	}
+
+	wr.lInstanceInfo = widgets.NewList()
+	wr.lChainInfo = widgets.NewList()
+}
+
+func (wr *WidgetsRender) setGrid() {
+	wr.grid.Set(
+		ui.NewRow(completRow/7, wr.pPublicKey),
```

Comment:
another magic numbers set: 7, 10, 3, 4, 2

Reply:
created local variables for this magic numbers.